packages/webdriverio/src/utils.js

```diff
@@ -81,8 +81,11 @@ export const findStrategy = function (value, isW3C, isMobile) {
         value = value.replace(/<|>|\/|\s/g, '')
 
     // use name strategy if value queries elements with name attributes for JSONWP
+    // or if isMobile is used even when w3c is used
     // e.g. "[name='myName']" or '[name="myName"]'
-    } else if (!isW3C && value.search(/^\[name=("|')([a-zA-z0-9\-_. ]+)("|')]$/) >= 0) {
+    } else if (isMobile
+        ? value.search(/^\[name=("|')([a-zA-z0-9\-_. ]+)("|')]$/) >= 0
+        : !isW3C && value.search(/^\[name=("|')([a-zA-z0-9\-_. ]+)("|')]$/) >= 0) {
```

Comment:
Can we move `value.search(/^\[name=("|')([a-zA-z0-9\-_. ]+)("|')]$/) >= 0` into a const somewhere upwards?

Reply:
Yep, I'm thinking about refactor this function in another PR.